lib/AST/Decl.cpp

```diff
@@ -1031,7 +1031,7 @@ unsigned PatternBindingDecl::getPatternEntryIndexForVarDecl(const VarDecl *VD) c
     ++Result;
   }
   
-  assert(0 && "PatternBindingDecl doesn't bind the specified VarDecl!");
+  llvm_unreachable("PatternBindingDecl doesn't bind the specified VarDecl!");
```

Comment:
This makes a small semantic difference: llvm_unreachable is an optimization hint in release builds, whereas `assert(0)` allows recovery. Some of these `assert(0)`s may therefore be intentional.